src/Microsoft.Extensions.Caching.Memory/MemoryCacheOptions.cs

```diff
@@ -2,22 +2,29 @@
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
 using System;
+using System.Collections.Generic;
 using Microsoft.Extensions.Internal;
 using Microsoft.Extensions.Options;
 
 namespace Microsoft.Extensions.Caching.Memory
 {
-    public class MemoryCacheOptions : IOptions<MemoryCacheOptions>
+    public class MemoryCacheOptions
     {
+        public MemoryCacheOptions(IEnumerable<IConfigureOptions<MemoryCacheOptions>> configureOptions = null)
```

Comment:
Wasn't immediately clear how we could get it type correctly, but I didn't spend much time on it, I only tried the following before moving on for now

```
public abstract class Options<TOptions> where TOptions : class
{
    public Options(IEnumerable<IConfigureOptions<TOptions>> configure = null) {
        if (configure != null) foreach (var c in configure) 
             // BLOWS UP: because this is not a TOptions
             c.Configure(this);
    }
}